provisioner/terraform/executor.go

```diff
@@ -242,6 +246,29 @@ func (e *executor) init(ctx, killCtx context.Context, logr logSink) error {
 	}
 
 	err := e.execWriteOutput(ctx, killCtx, args, e.basicEnv(), outWriter, errBuf)
+
+	// Check if .terraform.lock.hcl was modified after terraform init
+	postInitLockFileContent, _ := os.ReadFile(lockFilePath)
+	diff := generateFileDiff(preInitLockFileContent, postInitLockFileContent)
+	if diff != "" {
+		// Log informational message about lock file changes with diff
+		infoMsg := "INFO: .terraform.lock.hcl was modified during 'terraform init'. " +
+			"This is normal when Terraform downloads providers or updates dependencies. " +
+			"See https://developer.hashicorp.com/terraform/language/files/dependency-lock#understanding-lock-file-changes " +
+			"for more information about lock file changes."
```

Comment:
I would argue that this is better left at WARN. If the lockfile is being modified, this potentially means that Coder is needlessly downloading providers from the internet instead of using cached versions, which means longer workspace build times.